cv32e40s/tests/programs/custom/illegal_access_loop_tests/helpers.S

```diff
@@ -0,0 +1,57 @@
+/*
+**
+** Copyright 2022 OpenHW Group
+**
+** Licensed under the Solderpad Hardware Licence, Version 2.0 (the "License");
+** you may not use this file except in compliance with the License.
+** You may obtain a copy of the License at
+**
+** https://solderpad.org/licenses/
+**
+** Unless required by applicable law or agreed to in writing, software
+** distributed under the License is distributed on an "AS IS" BASIS,
+** WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+** See the License for the specific language governing permissions and
+** limitations under the License.
+**
+*******************************************************************************
+**
+** Assembly file which holds helper functions used in 'illegal_access_loop_tests.c' to declare pmp_regions or switch privilege_modes. 
+**
+*******************************************************************************
+*/
+
+.section .text
+
+.global setup_pmp
+.global set_u_mode
+
+setup_pmp:
+	// Set pmp addr to 0xFFFF_FFFF
+	li	t0, 0xFFFFFFFF
+	csrrw	x0, pmpaddr0, t0
+
+	// Set pmp region TOR and read/write/execute
+	li	t0, ((1 << 3) + (7 << 0))
+	csrrw	x0, pmpcfg0, t0
+
+	// Return to caller
+	jalr	x0, 0(ra)
+
+set_u_mode: // puts the core in usermode.
+	// setup the return adress to stack.
+	addi	sp, sp, -4
+	// store the ra to the stack pointer.
+	sw	ra, 0(sp)
```

Comment:
`ra` is saved on the stack, and later restored.
But is it ever changed in between?
It looks like the stack manipulation is not necessary, or I am missing something.

Reply:
No, I think this is old and also happens in the privilege_tests.S file. I think this is something I added when I had just started, but I agree, I don't understand what it does here and I don't think the ra gets changed either.

Reply:
If this function is used by the rest of the code then it would be nice to clean it up. (If it is not used then it should be removed.)